sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/LastValueAggregation.java

```diff
@@ -45,6 +46,11 @@ private LastValueAggregation() {}
     throw new IllegalArgumentException("Invalid instrument value type");
   }
 
+  @Override
+  public boolean isCompatibleWithInstrument(InstrumentDescriptor instrumentDescriptor) {
+    return instrumentDescriptor.getType() == InstrumentType.OBSERVABLE_GAUGE;
```

Comment:
this seems overly restrictive to me. Why not allow this for other instruments? There certainly won't be harm in it, will there?

Reply:
> There certainly won't be harm in it, will there?

I don't know, but here's where my head is at: Suppose the metrics SDK spec comes out with a table that expresses instrument - aggregation compatibility. Would it be a breaking change if we were to initially not restrict which instruments can use last value, and later add restrictions? Possibly.

I figure its easier to start off with a restrictive set of behaviors and later loosen things up, than it is to add new restrictions later.